gemd/entity/value/categorical_value.py

```diff
@@ -8,3 +11,16 @@ class CategoricalValue(BaseValue):
 
     All category names must be in unicode.
     """
+
+    @abstractmethod
+    def to_bounds(self) -> CategoricalBounds:
```

Comment:
Non-blocking comment: I think you can get this typing behavior by using a generic in `BaseValue` that would avoid needing to re-define the whole method and docstring.  Here's an example: https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/resources/data_concepts.py#L212

Reply:
I might be missing something, but I don't think that does quite what this is trying to do.  Happy to chat if you'd like to follow up.